src/test/fuzz/feefrac.cpp

```diff
@@ -121,3 +106,128 @@ FUZZ_TARGET(feefrac)
     assert((fr1 == fr2) == std::is_eq(cmp_total));
     assert((fr1 != fr2) == std::is_neq(cmp_total));
 }
+
+FUZZ_TARGET(feefrac_div_fallback)
+{
+    // Verify the behavior of FeeFrac::DivFallback over all possible inputs.
+
+    // Construct a 96-bit signed value num, a positive 31-bit value den, and rounding mode.
+    FuzzedDataProvider provider(buffer.data(), buffer.size());
+    auto num_high = provider.ConsumeIntegral<int64_t>();
+    auto num_low = provider.ConsumeIntegral<uint32_t>();
+    std::pair<int64_t, uint32_t> num{num_high, num_low};
+    auto den = provider.ConsumeIntegralInRange<int32_t>(1, std::numeric_limits<int32_t>::max());
+    auto round_down = provider.ConsumeBool();
+
+    // Predict the sign of the actual result.
+    bool is_negative = num_high < 0;
+    // Evaluate absolute value using arith_uint256. If the actual result is negative and we are
+    // rounding down, or positive and we are rounding up, the absolute value of the quotient is
+    // the rounded-up quotient of the absolute values.
+    auto num_abs = Abs256(num);
+    auto den_abs = Abs256(den);
+    auto quot_abs = (is_negative == round_down) ?
+        (num_abs + den_abs - 1) / den_abs :
+        num_abs / den_abs;
+
+    // If the result is not representable by an int64_t, bail out.
+    if ((is_negative && quot_abs > MAX_ABS_INT64) || (!is_negative && quot_abs >= MAX_ABS_INT64)) {
+        return;
+    }
+
+    // Verify the behavior of FeeFrac::DivFallback.
+    auto res = FeeFrac::DivFallback(num, den, round_down);
+    assert(res == 0 || (res < 0) == is_negative);
+    assert(Abs256(res) == quot_abs);
+
+    // Compare approximately with floating-point.
+    long double expect = round_down ? std::floorl(num_high * 4294967296.0L + num_low) / den
+                                    : std::ceill(num_high * 4294967296.0L + num_low) / den;
```

Comment:
It seems to me that `std::floor` and `std::ceil` already have overloads for `long double`, which we could use instead to appease the CI gods:

> feefrac.cpp:144:44: error: ‘floorl’ is not a member of ‘std’; did you mean ‘floor’?